app/Console/Commands/Optimize.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Foundation\Console\OptimizeCommand;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use function Safe\sprintf;
+use Symfony\Component\Console\Exception\ExceptionInterface as ConsoleException;
+use Symfony\Component\Console\Exception\InvalidOptionException;
+
+/**
+ * Improves the original "optimize" command provided by the framework.
+ *
+ * There are three improvements:
+ *
+ *  1) This command explicitly clears the old cache before building a new one.
+ *     Note that rebuilding a new cache is not sufficient, because this
+ *     does *not* overwrite the entire cache but some leftovers may remain.
+ *     This actually looks like an oversight and bug in the original command.
+ *  2) This command adds a "clever" mode which only rebuilds a new cache if
+ *     a previous cache has existed.
+ *     We use this in our install/update scripts in order to rebuild the cache
+ *     after installation/update without enforcing to use a cache for everyone.
+ *  3) This command adds a confirmation, if the user requests to build a cache
+ *     for non-productive environments as this is most likely an error and
+ *     undesired.
+ *     The confirmation can be skipped by pre-selecting the answer via a
+ *     command line option.
+ */
+class Optimize extends OptimizeCommand
+{
+	/**
+	 * The name and signature of the console command.
+	 *
+	 * @var string
+	 */
+	protected $signature = 'optimize
+		{--clever : Only (re-)creates cache if cache has already been created before and if not in production mode}
+		{--dont-confirm= : [assume-yes|assume-no] Don\'t ask for confirmation, but silently assume yes or no; "assume-yes" yields same behaviour as the original Laravel command}';
+
+	/**
+	 * Execute the console command.
+	 *
+	 * @return void
+	 *
+	 * @throws ConsoleException
+	 * @throws BindingResolutionException
+	 * @throws NotFoundExceptionInterface
+	 * @throws ContainerExceptionInterface
+	 */
+	public function handle(): void
+	{
+		$shallBeClever = $this->hasOption('clever') && $this->option('clever') === true;
+		$confirmationDefault = match ($this->option('dont-confirm')) {
+			'assume-yes' => true,
+			'assume-no' => false,
+			null => null,
+			default => throw new InvalidOptionException(sprintf('Unexpected option value %s for --dont-confirm', $this->option('dont-confirm')))
+		};
+		$hasPreviousCache = file_exists($this->laravel->getCachedConfigPath()) || file_exists($this->laravel->getCachedRoutesPath());
+
+		$this->call('optimize:clear');
+
+		if ($shallBeClever && !$hasPreviousCache) {
+			return;
+		}
+
+		if ($this->isNonProductive()) {
+			$this->alert('Application not in Production!');
```

Comment:
```suggestion
			$this->alert('Application not in production!');
```

Any particular reason for that capitalization?

Reply:
Yes, to be consistent with the original Laravel Artisan commands. See https://github.com/laravel/framework/blob/abd4787bcaf0bb55dd4be35bf15f8985ee06bbcc/src/Illuminate/Console/ConfirmableTrait.php#L16.

But I had the same thought as you.